bin/propolis-server/src/lib/initializer.rs

```diff
@@ -575,6 +577,59 @@ impl<'a> MachineInitializer<'a> {
         Ok(())
     }
 
+    pub fn initialize_test_devices(
```

Comment:
Do we want this behind a `not(omicron-build)` guard?

Reply:
Currently, when compiled with `omicron-build`, this function will skip initializing test devices and print a warning stating that they are not available in release builds:

https://github.com/oxidecomputer/propolis/blob/3705717b3ae3b8944631f6809783530cd49f3e9e/bin/propolis-server/src/lib/initializer.rs#L588-L595

We could change this function to not be compiled in at all when `omicron-build` is set, but we would no longer detect the presence of unsupported test devices and warn about them in `omicron-build` builds. If you'd prefer that, I can change it --- I just thought it was nice to warn if a test device was found in the config while running an omicron build. What do you think?

Reply:
I like the idea of avoiding the entire call to the init function, since it helps to avoid the sort of bug where we add another test device to `initialize_test_devices` but forget to add the `cfg` check. Another way to make this observable is to make the caller do the logging:

```rust
if cfg!(not(feature = "omicron-build")) {
    init.initialize_test_devices(...)?;
} else {
    info!("omicron-build was configured, won't enable test devices");
}
```

The latter call could be upgraded to something like `warn_if_test_devices_present`, but if I imagine being the audience of this message--I've added a test device to the Omicron package TOML, I'm testing with that package, it didn't work, so now I'm looking at the Propolis logs--I think I'd be satisfied with a log line during init that reminds me that Propolis doesn't initialize these for prod builds.

Reply:
@gjcolombo I'm fine with moving this to the caller, if you prefer that. My thinking was that it was kinda nice to only log about test devices if we actually encountered such a device in the config, rather than always logging that we won't enable them --- the information is really only relevant if there are test devices in the config, and if there aren't any, there's no reason to log it. But, I also see the value in compiling out the test device initialization entirely in prod builds. So, I'm fine with always logging.

Reply:
changed in 58ec187d81f77e1980edbda99645a49931fa2bfb